src/generic_layers.rs

```diff
@@ -72,20 +72,22 @@ impl<P: Reducible, const SIZE: u8, const SALT: u64> Debug for UniformPoint<P, SI
 
 fn generate_points<const SALT: u64, C: Chunk + 'static>(
     index: GridPoint<C>,
+    seed: u64,

```

Comment:
I think now that we have a struct for the seed we should use it everywhere and only get the integer out when actually seeding an rng